sql-native/src/main/java/org/ballerinalang/sql/nativeimpl/CallProcessor.java

```diff
@@ -59,18 +63,26 @@
 import static org.ballerinalang.sql.Constants.TYPE_DESCRIPTIONS_NATIVE_DATA_FIELD;
 import static org.ballerinalang.sql.utils.Utils.getColumnDefinitions;
 import static org.ballerinalang.sql.utils.Utils.getDefaultRecordType;
-import static org.ballerinalang.sql.utils.Utils.getOutParameterType;
-import static org.ballerinalang.sql.utils.Utils.setSQLValueParam;
+import static org.ballerinalang.sql.utils.Utils.getSqlQuery;
 import static org.ballerinalang.sql.utils.Utils.updateProcedureCallExecutionResult;
 
 /**
  * This class holds the utility methods involved with executing the call statements.
+ *
+ * @since 0.5.6
  */
-public class CallUtils {
-    private static final Calendar calendar = Calendar.getInstance(
-            TimeZone.getTimeZone(Constants.TIMEZONE_UTC.getValue()));
+public class CallProcessor {
+    private static final Calendar calendar = Calendar
+            .getInstance(TimeZone.getTimeZone(Constants.TIMEZONE_UTC.getValue()));
 
     public static Object nativeCall(BObject client, Object paramSQLString, BArray recordTypes) {
+        return nativeCall(client, paramSQLString, recordTypes, StatementParameterProcessor.getInstance(),
+            ResultParameterProcessor.getInstance());
+    }
+
+    public static Object nativeCall(BObject client, Object paramSQLString, BArray recordTypes, 
```

Comment:
Why do we need another method as this?  IMO, this logic can be moved to the original method above?

Reply:
Our intention was to,
- let the connector modules that do not override the Result & Statement ParameterProcessor classes to call,
 ```js 
public static Object nativeCall(BObject, Object, BArray);
```
 so that they can use the generic ParameterProcessor instances and,

- let the connector modules that override ParameterProcessor classes use,
```js
public static Object nativeCall(BObject , Object , BArray, StatementParameterProcessor,  ResultParameterProcessor);
```
 so that they can pass their connector specific ParameterProcessor instances.

Reply:
Lets add java doc comments if so, giving clear description

Reply:
Resolved with 43eb47e

Reply:
Resolved with [`a83a5d4`](https://github.com/ballerina-platform/module-ballerina-sql/pull/104/commits/a83a5d4f56474fabf2d465acd88319aa41f4e9c1)